### PR TITLE
Added missing edge-cases

### PR DESCRIPTION
**Pull request by ismael-miguel**
Discussion on: https://github.com/twigphp/Twig/issues/2677

### CODE REVIEW

File: doc/tags/if.rst [hunk 1]
```diff
@@ -68,7 +68,10 @@ use more complex ``expressions`` there too:
     ====================== ====================
     empty string           false
     numeric zero           false
+    NAN (Not A Number)     true
+    INF (Infinity)         true
     whitespace-only string true
+    string "0" or '0'      false
```
Comment (stof) [inline doc/tags/if.rst:74]: that's the same string. No need for this `or`

Comment (ismael-miguel) [reply doc/tags/if.rst:74]: PHP's single-quoted and double-quoted strings behave differently.

Doing `"\t"` will show a tab while `'\t'` will show `\t`.
But in this case, yes, it's the same thing.
Just over-clarification.

Comment (stof) [reply doc/tags/if.rst:74]: but that's about the *syntax* of the language. Once you have a string, there is no difference about how it was initially quoted. And for `0`, there is no difference

Comment (ismael-miguel) [reply doc/tags/if.rst:74]: Quoting my previous comment:
```
But in this case, yes, it's the same thing.
Just over-clarification.
```

**Comment by fabpot** [comment]
Thank you @ismael-miguel.

**Pull request merged**